Flint/src/main/java/flint/CLIAdapter.java

```diff
@@ -1,65 +1,193 @@
+    }
+
+    String realConfigPath = configPath;
+
+    if (realConfigPath.charAt(0) != '\\' && realConfigPath.charAt(0) != '/') {
+      // if not passing absolute path
+      realConfigPath = "file://" + System.getProperty("user.dir") + "/" + realConfigPath;
+    } else {
+      realConfigPath = "file://" + realConfigPath;
+    }
+
+    URLClassLoader urlClassLoader = URLClassLoader.newInstance(new URL[] {
+            new URL(
+                    realConfigPath
+            )
+    });
+
+    Class outputClass = urlClassLoader.loadClass(className);
+
+    if (!outputClass.getSuperclass().getSimpleName().equals("FlintConfiguration")) {
+      throw new IllegalArgumentException();
     }
 
-    /**
-     * Starts the Driver to run Flint on fileToLint with config.
-     *
-     * @param fileToLint - The file path that leads to a valid and existing Java file.
-     * @param config - The configuration to run on flieToLint.
-     * */
-    private static Collection<LintFailure> run(String fileToLint, FlintConfiguration config) throws Exception{
-        throw new Exception("Not Yet Implemented");
+    FlintConfiguration output = (FlintConfiguration)outputClass.newInstance();
+
+    return output;
+  }
+
+  /**
+  * This function checks if filePath leads to an existing java file.
+  *
+  * @param filePath - The file path to check.
+  * @return Returns the true if the file exists and is a java file. Otherwise false.
+  * */
+  public static boolean isJava(String filePath) {
+    File file = new File(filePath);
+
+    return Pattern.matches(".*\\.java$", filePath) && file.exists();
+  }
+
+  /**
+  * Starts the Driver to run Flint on fileToLint with config.
+  *
+  * @param fileToLint - The file path that leads to a valid and existing Java file.
+  * @param config - The configuration to run on flieToLint.
+  * // Otherexception that will be returned from run
+  * */
+  public static Collection<LintFailure> run(String fileToLint, FlintConfiguration config) throws Exception{
+    return FlintDriver.run(fileToLint, config);
+  }
+
+  /**
+  * Outputs the result of the Flint check.
+  *
+  * @require result should not be null
+  * @param result - The collection of failure results from Driver
+  * @throws NullPointerException - When result is null
+  * */
+  public static void resultOutput(Collection<LintFailure> result) throws NullPointerException{
+    if (result == null) {
+      throw new NullPointerException();
     }
 
-    /**
-     * Outputs the result of the Flint check.
-     *
-     * @require result should not be null
-     * @param result - The collection of failure results from Driver
-     * @throws NullPointerException - When result is null
-     * // Otherexception that will be returned from run
-     * */
-    private static void resultOutput(Collection<LintFailure> result) throws Exception{
-        throw new Exception("Not Yet Implemented");
+    Iterator<LintFailure> failureIt = result.iterator();

```

Comment:
Why not use a for-each loop?
Something like `for (LintFailure failure : result) { }`